src/pages/LoginHomepage/LoginHomepage.jsx

```diff
@@ -1,48 +1,64 @@
-import { useNavigate } from 'react-router-dom';
-import './loginHomepage.css';
+import { useNavigate } from "react-router-dom";
+import "./loginHomepage.css";
+import { useState } from "react";
 
 export default function LoginHomepage() {
   const navigate = useNavigate();
+  const [user, setUser] = useState({
+    username: "",
+  });
+
+  function handleUserChange(e) {
+    const newUsername = e.target.value;
+    setUser((prevUser) => ({ ...prevUser, username: newUsername }));
+  }
 
   function handleLogin(e) {
     e.preventDefault();
-    navigate('/articles');
+    localStorage.setItem("username", user.username);
+    navigate("/articles");
```

Comment:
This function should instead make use of context within React. You can use context here to provide the setUsername function to your entire application. This ideally is done from the highest point of your component tree, the App component.

Using context at the highest point we can then just grab the setUsername function from anywhere within the application to get around this issue of storing locally to grab it somewhere else

That might look like

```js

// App.jsx

const [username, setUsername] = useState("")
<Context.provider value={setUsername}>
// rest of your components
</Context.provider>

// LoginHomepage.jsx
const {setUsername} = useContext(UserContext)
```

If you feel that this will cause more problems for the moment then come back to this at a later stage :)